File: 백준/Silver/2417. 정수 제곱근/정수 제곱근.java
import java.util.*;
import java.util.stream.Collectors;

public class Main {

    public static void main(String[] args) throws Exception {

        Scanner sc = new Scanner(System.in);

        long n = sc.nextLong();

        long start = 0;
        long end = n;

        // 어떤수의 제곱이 n보다 크거나 같은, 가장 작은 정수

        while (start < end) {

            long mid = (start + end) / 2;

            if (Math.pow(mid, 2) < n) {
                start = mid + 1;
            } else { // 더 작다
                end = mid;
            }
        }

        System.out.println(start);
    }
}

File: 백준/Silver/2417. 정수 제곱근/정수 제곱근.py
n = int(input().strip())

start = 0
end = n  # [start, end) 범위

# 어떤 수의 제곱이 n보다 크거나 같은, 가장 작은 정수
while start < end:
    mid = (start + end) // 2

    if mid * mid < n:
        start = mid + 1
    else:  # mid*mid >= n
        end = mid

print(start)